### PR TITLE
Documentation updates in preparation for code pattern

### PR DESCRIPTION
**Pull request by ajbozarth**
- Moved doc images to standard location
- Added standard doc files
- Updated docker hub references to use new codait repo
- Various style and grammar updates to README
- Added missing sections to README
- Added troubleshooting for python version error

### CODE REVIEW

**Comment by ajbozarth** [comment]
My code pattern related review are contained in this PR and the issues I've added/updated. I did not do any review of the code and only played with UI to make sure it worked ok. I'll trust @ptitzler and @kastentx that the actual app is solid from a code perspective. My PR/Issues are all documentation related.

I did run into issues trying to start the app using `npm` and once I can figure out why it doesn't work we might want to add a troubleshooting or extra install step to make sure others don't hit what I'm hitting.

**Comment by ajbozarth** [comment]
@rhagarty Nick has "finished" this application and as of this PR we're now preparing it as a code pattern. Feel free to start doing general review and making sure things are up to snuff. Nick is out until Tuesday so we can discuss things further at our meeting next Tuesday.

**Comment by ajbozarth** [comment]
I just pushed an extra update that consolidates the troubleshooting and adds a trouble shooting for the issue I ran into.

My issue was that my local version of python was 3.x and one of this app's dependancies requires python2.7. The solution to this was to run a `npm` command to point that library to `python2.7`

File: README.md [hunk 1]
```diff
@@ -78,118 +83,143 @@ Go to `http://localhost:5000` to load it. From there you can explore the API and
...
 
-First, clone this repo with the command: 
+First, clone this repo with the command:
 ```bash
-git clone https://github.com/IBM/MAX-Image-Segmenter-Web-App.git
+$ git clone https://github.com/IBM/MAX-Image-Segmenter-Web-App.git
 ```
 
-Enter the directory with `cd`, then install dependencies with the command: 
+Enter the directory with `cd`, then install dependencies with the command:
```
Comment (ptitzler) [inline README.md]: "Install the dependencies." (goes together with comment above)

File: README.md [hunk 2]
```diff
@@ -78,118 +83,143 @@ Go to `http://localhost:5000` to load it. From there you can explore the API and
...
-
-<hr>
-<div style="text-align: center;">
+* [Starting the App](#starting-the-app)
+* [Uploading Images](#uploading-images)
+* [MAX Image Segmenter Response](#max-image-segmenter-response)
+* [Loading Images into the Studio](#loading-images-into-the-studio)
+* [Combining Objects to Create New Images](#combining-objects-to-create-new-images)
 
 #### Starting the App
```
Comment (ptitzler) [inline README.md]: General comment: The "sample output" section would  be easier to consume if an instruction preceeds the associated image.
Current flow example:
 Starting the App
   [screen capture]
    "This is the screen you will see upon first starting the app. Click "Add an Image" in the lower part of the screen to begin."

Proposed sequence:
 Starting the App
    "This is the screen you will see upon first starting the app. Click "Add an Image" in the lower part of the screen to begin."
   [screen capture to which the instruction applies]

File: README.md [hunk 3]
```diff
@@ -1,32 +1,46 @@
...
+In this application, the [MAX Image Segmenter](https://github.com/IBM/MAX-Image-Segmenter) model is used to identify the objects in a user-submitted image on a pixel-by-pixel level. These categorized pixels are then used to generate a version of the image with each unique type of object highlighted in a separate color, called a colormap. Each segment is then split into its own image file which can be downloaded for use elsewhere. As subsequent images are uploaded, they will be added to the carousel in the lower portion of the screen and saved in the browser, using PouchDB. From this carousel, images can be reviewed, deleted, or loaded into the "Studio".
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: For code patterns there is a specific screen shots section towards the bottom, I believe they wouldn't like images this high up in the README. (And Nick has great screenshots of this below). But maybe we could add a link to that section somewhere here instead

File: README.md [hunk 4]
```diff
@@ -1,32 +1,46 @@
...
 
 In the Studio section of the app, two images may be loaded into an interface that allows for drag-and-drop combinations of any two objects within them. Any new images you may happen to create here can also be downloaded.
 
 When the reader has completed this Code Pattern, they will understand how to:
 
-* Build a docker image of the Image Segmenter MAX Model
 * Deploy a deep learning model with a REST endpoint
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: though what you suggested is not wrong, what Nick has here is consistent with all our previous code patterns READMEs

File: README.md [hunk 5]
```diff
@@ -1,32 +1,46 @@
...
 * Recognize objects segments in an image using the MAX Model's REST API
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: Again you're not wrong, but these bullets are consistent with our previous code patterns and I think we should stay consistent

File: README.md [hunk 6]
```diff
@@ -1,32 +1,46 @@
...
-* Run a web application that using the model's REST API
+* Run a web application that uses the model's REST API
 * Interact with processed object segments to create new images
 
-<br>
-<div style="text-align: center;">
-  <img src="./static/arch.png" alt="Architectural Diagram" />
-</div>
+![Architecture](doc/source/images/architecture.png)
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: @kastentx this update to the arch diagram is on you

File: README.md [hunk 7]
```diff
@@ -1,32 +1,46 @@
...
-3. Web app uses JSON response to crop objects from image and display them to user
-4. User interacts with object segments from uploaded images using web app UI
+1. User submits an image using the Web App UI
+2. MAX Model API processes the image and returns a JSON response
+3. Web App Server uses JSON response to crop objects from image and display them to user
+4. User interacts with object segments from uploaded images using Web App UI
+
+## Included Components
+
+* [IBM Model Asset Exchange](https://developer.ibm.com/code/exchanges/models/): A place for developers to find and use free and open source deep learning models.
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: it will redirect, that's my job :)

**Comment by ajbozarth** [comment]
@ptitzler a general response to the Model API vs DL microservice comments. Given the consistent use of the former in previous patterns this would be a change in terminology. As such I talked to Fred about it and we think we should discuss it as a team on Tuesday. Since both are technically correct we should makes sure as a team we're using a consistent terminology that the majority of us agree on (not just what one project leader thinks is best). Given your advocacy experience DL micro service could be the better usage overall.

File: README.md (hunk 2 above)
Comment (ajbozarth) [reply README.md]: ah this makes sense, it was probably an artifact of me moving this section out of steps into the screen shot section. I'll swap the img and text lines in this section

**Comment by ajbozarth** [comment]
@ptitzler I've addressed your review comments other that the ones mentioned in my previous comment if you'd like to take another look

**Comment by ptitzler** [comment]
Fair enough. I agree that both terms _can_ mean the same things. I favored MS because it is a "modern" pattern implying lightweight and flexible deployment/devops, whereas Model API does not imply anything positive (from a developers point of view).

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ptitzler) [reply README.md]: Dunno, that'd feel wrong, but I hear what you are saying.

File: README.md (hunk 2 above)
Comment (ptitzler) [reply README.md]: Good flow now. Thanks!

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajbozarth) [reply README.md]: @rhagarty opinions?

**Comment by ajbozarth** [comment]
@kastentx I want to hold off on merging this with the micro service changes until after scrum today. We plan to discuss verbiage in that meeting (API server vs microservice). Once that discussion is complete I'll either update this match or merge it

**Comment by ajbozarth** [comment]
Merging with `microservice` per the decision at scrum today

**Pull request merged**